web-app-router/scenes/Portal/Teams/TeamId/Team/ApiKeys/page/ApiKeyTable/ApiKeyRow/index.tsx

```diff
@@ -0,0 +1,177 @@
+"use client";
+import { formatDistanceToNowStrict } from "date-fns";
+import { TYPOGRAPHY, Typography } from "@/components/Typography";
+import { FetchKeysQuery } from "../../graphql/client/fetch-keys.generated";
+import {
+  Dropdown,
+  DropdownButton,
+  DropdownItem,
+  DropdownItems,
+} from "@/components/Dropdown";
+import { MoreVerticalIcon } from "@/components/Icons/MoreVerticalIcon";
+
+import { Status } from "./Status";
+import { EditIcon } from "@/components/Icons/EditIcon";
+import { KeyIcon } from "@/components/Icons/KeyIcon";
+import { TrashIcon } from "@/components/Icons/TrashIcon";
+import clsx from "clsx";
+import { useCallback, useMemo, useState } from "react";
+import { useResetApiKeyMutation } from "./graphql/client/reset-api-key.generated";
+import { toast } from "react-toastify";
+import { DecoratedButton } from "@/components/DecoratedButton";
+import { useUser } from "@auth0/nextjs-auth0/client";
+import { Auth0SessionUser } from "@/lib/types";
+import { Role_Enum } from "@/graphql/graphql";
+
+export const ApiKeyRow = (props: {
+  apiKey: FetchKeysQuery["api_key"][0];
+  index: number;
+  teamId: string;
+  openViewDetails: (key: FetchKeysQuery["api_key"][0]) => void;
+  openDeleteKeyModal: (key: FetchKeysQuery["api_key"][0]) => void;
+}) => {
+  const { apiKey, index, teamId, openViewDetails, openDeleteKeyModal } = props;
+  const [secretKey, setSecretKey] = useState<string | null>(null);
+  const timeAgo = formatDistanceToNowStrict(new Date(apiKey.created_at), {
+    addSuffix: true,
+  });
+  const { user } = useUser() as Auth0SessionUser;
+
+  const isEnoughPermissions = useMemo(() => {
+    const membership = user?.hasura.memberships.find(
+      (m) => m.team?.id === teamId,
+    );
+    return membership?.role === Role_Enum.Owner;
+  }, [teamId, user?.hasura.memberships]);
+
+  const [resetApiKeyMutation, { loading }] = useResetApiKeyMutation({
+    context: { headers: { team_id: teamId } },
+  });
+
+  const copyKey = useCallback(() => {
+    navigator.clipboard.writeText(secretKey ?? "");
+    toast.success("Copied to clipboard");
+  }, [secretKey]);
+
+  const resetAPIKey = useCallback(
+    async (apiKeyId: string) => {
+      if (loading) {
+        return;
+      }
+      try {
+        const result = await resetApiKeyMutation({
```

Comment:
@andy-t-wang I Just noticed that if the API key is inactive and you are trying to reset it you will get the: 
`User does not have sufficient permissions.` error which leads to a toast error. It can be confusing I think. Maybe we should handle it somehow to display toast with something like "Activate key to reset it"